### PR TITLE
Fixes TreeItem minimum width calculations

### LINKED ISSUE #84015: Long-named nodes aren't fully revealed when scrolling horizontally in the scene tree
### Godot version

v4.2.beta.custom_build [46cb7f94f]

### System information

Godot v4.2.beta (46cb7f94f) - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2070 (NVIDIA; 31.0.15.4584) - Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 Threads)

### Issue description

In the scene tree, when a node has a long name that doesn't fit in the width it is truncated (or clipped.)

Scrolling horizontally all the way to the right should reveal the whole name but it doesn't.

https://github.com/godotengine/godot/assets/1632314/a075f64a-aa43-450c-b1c6-37502d1c613e

### Steps to reproduce

Create a new scene.
Give the root node a long name.
Narrow the scene tree window, see that the name is truncated.
Scroll horizontally all the way to the right, see that you can't see the whole name and it is still truncated.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by chybby**
1. Adds missing margins to `TreeItem::get_minimum_size`.
2. Sets minimum size dirty flags when an item is (un)collapsed.
3. Fixes depth calculation in `Tree::get_column_minimum_width` that was using the next visible item's depth when calculating the current item's minimum width.

Current:

https://github.com/godotengine/godot/assets/1632314/02f1d6c3-8da9-40c9-a1c5-097a90f9ab04

After:

https://github.com/godotengine/godot/assets/1632314/97f3cbf2-bf5a-4cd3-b3a0-e0eb3a3de82b

Fixes #84015.

### CODE REVIEW

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -1433,6 +1442,14 @@ Size2 TreeItem::get_minimum_size(int p_column) {
 	if (cell.cached_minimum_size_dirty) {
 		Size2 size;
 
+		size.width += parent_tree->theme_cache.inner_item_margin_left + parent_tree->theme_cache.inner_item_margin_right;
+
+		if (p_column == 0) {
```
Comment (AThousandShips) [inline scene/gui/tree.cpp:1447]:
```suggestion
		if (p_column == 0 && !(disable_folding || parent_tree->hide_folding)) {
```

File: scene/gui/tree.cpp [hunk 2]
```diff
@@ -1433,6 +1442,14 @@ Size2 TreeItem::get_minimum_size(int p_column) {
...
+			size.width += (disable_folding || parent_tree->hide_folding) ? parent_tree->theme_cache.h_separation : parent_tree->theme_cache.item_margin;
```
Comment (AThousandShips) [inline scene/gui/tree.cpp:1448]:
```suggestion
			size.width += parent_tree->theme_cache.item_margin;
```

File: scene/gui/tree.cpp (hunk 1 above)
Comment (chybby) [reply scene/gui/tree.cpp]: Done

File: scene/gui/tree.cpp (hunk 2 above)
Comment (chybby) [reply scene/gui/tree.cpp]: Done

**Comment by chybby** [comment]
Just checked this was still an issue and synced to head.

**Comment by chybby** [comment]
Superseded by #91095

**Pull request closed without merging**